.github/workflows/automerge.yml

```diff
@@ -0,0 +1,17 @@
+name: Auto Merge
+on:
+  check_suite:
+    types:
+      - completed
+  status: {}
```

Comment:
Just to confirm, will this action run for all pull request every time all CI builds pass? Does this also run on pushes `main`?

Reply:
This action should just run on completed checks. In order to auto merge the PR if appropriate.